compose/ui/ui-text/src/skikoMain/kotlin/androidx/compose/ui/text/SkiaParagraph.skiko.kt

```diff
@@ -254,6 +257,19 @@ internal class SkiaParagraph(
         }
     }
 
+    private fun getAlignedStartingPosition(isRtl: Boolean) = when {
+        textAlign == TextAlign.Center -> width / 2
+        isRtl && textAlign == TextAlign.Start -> width
+        isRtl && textAlign == TextAlign.Right -> width
+        isRtl && textAlign == TextAlign.Left -> 0f
+        isRtl && textAlign == TextAlign.End -> 0f
+        !isRtl && textAlign == TextAlign.Start -> 0f
+        !isRtl && textAlign == TextAlign.Right -> width
+        !isRtl && textAlign == TextAlign.Left -> 0f
+        !isRtl && textAlign == TextAlign.End -> width
+        else -> 0f
```

Comment:
This can be simplified (and made more intuitive):

```
    private fun getAlignedStartingPosition(textAlign: TextAlign, isRtl: Boolean) = when (textAlign) {
        TextAlign.Center -> width / 2
        TextAlign.Left -> 0f
        TextAlign.Right -> width
        TextAlign.Start -> if (isRtl) width else 0f
        TextAlign.End -> if (isRtl) 0f else width
        else -> 0f
    }
```